pennylane/wires.py

```diff
@@ -15,19 +15,47 @@
 This module contains the :class:`Wires` class, which takes care of wire bookkeeping.
 """
 from collections import Sequence, Iterable
+import numpy as np
 from numbers import Number
-import numpy as np  # for random functions
 
 
 class WireError(Exception):
     """Exception raised by a :class:`~.pennylane.wires.Wire` object when it is unable to process wires.
     """
 
 
+def _process(wires):
+    """Converts the input to a tuple of numbers or strings."""
+
+    if isinstance(wires, Wires):
+        # if input is already a Wires object, just copy its wire tuple
+        return wires.wire_tuple
+
+    elif isinstance(wires, Iterable) and all(isinstance(w, Wires) for w in wires):
+        # if the elements are themselves Wires objects, merge them to a new one
+        wires = [w for wires_ in wires for w in wires_.tolist()]
+        return tuple(wires)
```

Comment:
Not sure how much it saves, but you can also do things like `tuple(a for a in b)` in python and it will avoid creating a list

Reply:
Good idea